mainchain-follower/db-sync-follower/src/candidates/datum/permissioned.rs

```diff
@@ -44,34 +42,36 @@ impl From<PermissionedCandidateDatums> for Vec<RawPermissionedCandidateData> {
 }
 
 fn decode_legacy_permissioned_candidates_datums(
-	datum: &Datum,
+	datum: PlutusData,
 ) -> super::Result<Vec<PermissionedCandidateDatumV0>> {
-	let permissioned_candidates: super::Result<Vec<PermissionedCandidateDatumV0>> = match datum {
-		ListDatum(list_datums) => list_datums
-			.iter()
-			.map(|keys_datum| match keys_datum {
-				ListDatum(d) => {
-					let sc = d.first().and_then(|d| d.as_bytestring())?;
-					let aura = d.get(1).and_then(|d| d.as_bytestring())?;
-					let grandpa = d.get(2).and_then(|d| d.as_bytestring())?;
-					Some(PermissionedCandidateDatumV0 {
-						sidechain_public_key: SidechainPublicKey(sc.clone()),
-						aura_public_key: AuraPublicKey(aura.clone()),
-						grandpa_public_key: GrandpaPublicKey(grandpa.clone()),
-					})
-				},
-				_ => None,
-			})
-			.collect::<Option<Vec<PermissionedCandidateDatumV0>>>(),
-		_ => None,
-	}
-	.ok_or(Box::new(DatumDecodeError {
+	let list_datums: PlutusList = datum.as_list().ok_or(DatumDecodeError {
 		datum: datum.clone(),
 		to: "PermissionedCandidateDatumV0".to_string(),
-	}));
+	})?;
 
-	if permissioned_candidates.is_err() {
-		error!("Could not decode {:?} to Permissioned candidates datum. Expected [[ByteString, ByteString, ByteString]].", datum.clone());
-	}
-	permissioned_candidates
+	let permissioned_candidates: Vec<PermissionedCandidateDatumV0> = list_datums
+		.into_iter()
+		.map(|keys_datum| match keys_datum.as_list() {
```

Comment:
I think it might be worth considering to extract this into a function such that you iterate over a `datum.as_list()` and call `decode_permissioned_candidate_datum` or whatever you want to call it?

Reply:
Added a helper function. We can revisit the errors on another occasion. I don't want to spend too much time on this step.